### PR TITLE
Remove redundant py2 helper code

### PR DESCRIPTION
**Pull request by alexander-beedie**
Some long-unnecessary py2 → py3 `int` and `str` helper stubs were lurking in the codebase. 

These can be removed to clean things up a bit (with the minor/bonus side-effect of making all the `isinstance` checks that referenced them ~1.5-2x faster, as the helper stubs were both single-element tuples).

### CODE REVIEW

File: datashader/datashape/predicates.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def isscalar(ds):
     """
     if isinstance(ds, str):
         ds = dshape(ds)
-    if isinstance(ds, DataShape) and len(ds) == 1:
+    elif isinstance(ds, DataShape) and len(ds) == 1:
```
Comment (Hoxbro) [inline datashader/datashape/predicates.py]: Are you sure you can do this? For example, If the calculation above makes ds a DataShape with length 1.

File: datashader/datashape/coretypes.py [hunk 2]
```diff
@@ -376,10 +372,10 @@ class String(Unit):
...
             fixlen, encoding = None, None
-        if len(args) == 1:
-            if isinstance(args[0], _strtypes):
+        elif len(args) == 1:
+            if isinstance(args[0], str):
                 fixlen, encoding = None, args[0]
-            if isinstance(args[0], _inttypes):
+            if isinstance(args[0], int):
                 fixlen, encoding = args[0], None
         if len(args) == 2:
```
Comment (Hoxbro) [inline datashader/datashape/coretypes.py:380]:
```suggestion
        elif len(args) == 2:
```

To align with changes above.

**Review by Hoxbro — COMMENTED** [review]
Thank you!

We vendored datashape this fall as it failed to install with Python 3.12, as it hadn't been updated for some time... That does not mean we should have Python 2 code in it. 

I haven't set over the test because https://github.com/holoviz/datashader/pull/1314 is needed for them to pass. I will set them over when the PR is merged.

File: datashader/datashape/predicates.py (hunk 1 above)
Comment (Hoxbro) [reply datashader/datashape/predicates.py]: You reverted this before I could press the review button, and then it disappeared for me.

Comment (alexander-beedie) [reply datashader/datashape/predicates.py]: Yes, I realised I had been a little over-enthusiastic once I got some of the tests working ;)

**Comment by alexander-beedie** [comment]
> We vendored datashape this fall as it failed to install with Python 3.12, as it hadn't been updated for some time... That does not mean we should have Python 2 code in it.

Makes sense! I was just starting to very gently poke around to see what it might take to get first-class Polars integration 🤔

File: datashader/datashape/predicates.py (hunk 1 above)
Comment (Hoxbro) [reply datashader/datashape/predicates.py]: I am very familiar with that feeling.

**Comment by Hoxbro** [comment]
Thank you for the PR :+1:

**Pull request merged**